src/BuiltInTools/dotnet-watch/Program.cs

```diff
@@ -59,6 +61,13 @@ public class Program : IDisposable
 
         public Program(IConsole console, string workingDirectory)
         {
+            // We can register the MSBuild that is bundled with the SDK to perform MSBuild things. dotnet-watch is in
+            // a nested folder of the SDK's root, we'll back up to it.
+            // AppContext.BaseDirectory = $sdkRoot\$sdkVersion\DotnetTools\dotnet-watch\$version\tools\net6.0\any\
+            // MSBuild.dll is located at $sdkRoot\$sdkVersion\MSBuild.dll
+            var sdkRootDirectory = Path.Combine(AppContext.BaseDirectory, "..", "..", "..", "..", "..", "..");
```

Comment:
Is there no way to get `sdkRoot` and `sdkVersion` directly without having to travel up the directory tree like this?

Reply:
I couldn't find anything obvious. You can get to the path of the dotnet.exe and the shared framework, but nothing obvious about this thing. Note that the opposite of this code is already baked in to the CLI today: https://github.com/dotnet/sdk/blob/main/src/Cli/dotnet/CommandFactory/CommandResolution/DotnetToolsCommandResolver.cs#L39-L43